users.js

```diff
@@ -559,9 +655,11 @@ class User {
 	 * special permission check function is required to carry out this check
 	 * because we need to know which socket the client is connected from in
 	 * order to determine the relevant IP for checking the whitelist.
+	 * @param {Connection} connection
 	 */
 	hasConsoleAccess(connection) {
 		if (this.hasSysopAccess()) return true;
+		// @ts-ignore
```

Comment:
Wait, why this?

Reply:
w/o the ts-ignore I have to pass null parameters into `this.can`: would that be preferred (here and everywhere else with this.can)?

Reply:
Why not just let `this.can` accept default parameters?

Reply:
...forgot that was a thing